### PR TITLE
Add google analytics

### PR DESCRIPTION
**Pull request by jonathanp727**
Added new component which links react router changes with google analytics.

Test plan:  Ran dev-server and made sure there were no errors.

### CODE REVIEW

File: source/App.jsx [hunk 1]
```diff
@@ -1,20 +1,27 @@
 // @flow
 import React from 'react';
 import { BrowserRouter, Switch, Route } from 'react-router-dom';
+import ReactGA from 'react-ga';
 
 import Error from 'scenes/Error';
 import Home from 'scenes/Home';
 import Sponsor from 'scenes/Sponsor';
 
+import { RouteTracker } from './utils/GoogleAnalytics';
```
Comment (jc3m) [inline source/App.jsx]: Change `utils` to `services`

File: source/utils/GoogleAnalytics.jsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      logPageChange(pathname, search);
+    }
+  }
+
+  render() {
+    return null;
+  }
+}
+
+GoogleAnalytics.propTypes = {
```
Comment (jc3m) [inline source/utils/GoogleAnalytics.jsx]: Use flow types for Props instead of PropTypes

**Comment by jonathanp727** [comment]
Test plan: Tested by adding console.log statement to logPageChange() function and made sure it printed.

#26

File: source/services/GoogleAnalytics.jsx [hunk 3]
```diff
@@ -1,8 +1,14 @@
 import React, { Component } from 'react';
-import PropTypes from 'prop-types';
 import ReactGA from 'react-ga';
 import { Route } from 'react-router-dom';
 
+type props = {
```
Comment (jc3m) [inline source/services/GoogleAnalytics.jsx]: Convention is capitalize types `props` should be `Props`

File: source/services/GoogleAnalytics.jsx [hunk 4]
```diff
@@ -1,8 +1,14 @@
...
+  location: {
```
Comment (jc3m) [inline source/services/GoogleAnalytics.jsx]: I believe React Router should have a predefined type for the location prop. Check out https://github.com/HackIllinois/site-2019/blob/staging/flow-typed/npm/react-router-dom_v4.x.x.js#L51

**Pull request merged**